app/src/androidTest/java/com/karumi/katasuperheroes/MainActivityTest.java

```diff
@@ -65,6 +85,70 @@
     onView(withText("¯\\_(ツ)_/¯")).check(matches(isDisplayed()));
   }
 
+  @Test public void showsEmptyCaseWhenThereAreSuperHeroes(){
+    givenThereAreOneSuperHeroe();
+    startActivity();
+    onView(withText("¯\\_(ツ)_/¯")).check(matches(not(isDisplayed())));
+  }
+
+  @Test public void viewShowsSuperHeroeName(){
+    givenThereAreOneSuperHeroe();
+    startActivity();
+    onView(withId(R.id.tv_super_hero_name)).check(matches(isDisplayed()));
+  }
+
+  @Test public void showSuperHeroesNameIfThereAreSuperHeroes(){
+    final List<SuperHero> superHeroes = giverNSuperHeroes(100);
+
+    startActivity();
+
+    RecyclerViewInteraction.<SuperHero>onRecyclerView(withId(R.id.recycler_view))
+            .withItems(superHeroes)
+            .check(new RecyclerViewInteraction.ItemViewAssertion<SuperHero>() {
+              @Override
+              public void check(SuperHero item, View view, NoMatchingViewException e) {
+                matches(hasDescendant(withText(item.getName()))).check(view,e);
+              }
+            });
+  }
+
+  @Test public void tapPushedOnOneSuperHeroe(){
+    final List<SuperHero> superHeroes = giverNSuperHeroes(100);
+    int index = 0;
+    startActivity();
+
+    onView(withId(R.id.recycler_view))
+            .perform(RecyclerViewActions.actionOnItemAtPosition(index,click()));
+    SuperHero superHeroeSelected = superHeroes.get(index);
+    intended(hasComponent(SuperHeroDetailActivity.class.getCanonicalName()));
+    intended(hasExtra("super_hero_name_key",superHeroeSelected.getName()));
+    ToolbarMatcher.onToolbarWithTitle(superHeroeSelected.getName());
+
+
+  }
```

Comment:
This test is failing because you are not mocking the ``getByName`` call in your ``SuperHeroesRepository`` mocked instance.